test/Microsoft.Health.Fhir.Tests.Integration/Features/Operations/Export/CreateExportRequestHandlerTests.cs

```diff
@@ -0,0 +1,151 @@
+
+            _createExportRequestHandler = new CreateExportRequestHandler(_claimsExtractor, _fhirOperationDataStore, _secretStore);
+        }
+
+        public Task InitializeAsync()
+        {
+            return _fhirStorageTestHelper.DeleteAllExportJobRecordsAsync();
+        }
+
+        public Task DisposeAsync()
+        {
+            return Task.CompletedTask;
+        }
+
+        [Fact]
+        public async Task GivenThereIsNoMatchingJob_WhenCreatingAnExportJob_ThenNewJobShouldBeCreated()
+        {
+            var request = new CreateExportRequest(RequestUrl, DestinationType, ConnectionString);
+
+            CreateExportResponse response = await _createExportRequestHandler.Handle(request, _cancellationToken);
+
+            Assert.NotNull(response);
+            Assert.NotEmpty(response.JobId);
+
+            SecretWrapper secret = await _secretStore.GetSecretAsync($"Export-Destination-{response.JobId}");
+
+            Assert.NotNull(secret);
+        }
+
+        [Fact]
+        public async Task GivenThereIsAMatchingJob_WhenCreatingAnExportJob_ThenNewJobShouldBeCreated()
+        {
+            var request = new CreateExportRequest(RequestUrl, DestinationType, ConnectionString);
+
+            CreateExportResponse response = await _createExportRequestHandler.Handle(request, _cancellationToken);
+
+            var newRequest = new CreateExportRequest(RequestUrl, DestinationType, ConnectionString);
+
+            CreateExportResponse newResponse = await _createExportRequestHandler.Handle(request, _cancellationToken);
+
+            Assert.NotNull(newResponse);
+            Assert.Equal(response.JobId, newResponse.JobId);
+        }
+
+        [Fact]
+        public async Task GivenDifferentRequestUrl_WhenCreatingAnExportJob_ThenNewJobShouldBeCreated()
+        {
+            var request = new CreateExportRequest(RequestUrl, DestinationType, ConnectionString);
+
+            CreateExportResponse response = await _createExportRequestHandler.Handle(request, _cancellationToken);
+
+            var newRequest = new CreateExportRequest(new Uri("http://localhost/test"), DestinationType, ConnectionString);
+
+            CreateExportResponse newResponse = await _createExportRequestHandler.Handle(newRequest, _cancellationToken);
+
+            Assert.NotNull(newResponse);
+            Assert.NotEqual(response.JobId, newResponse.JobId);
+        }
+
+        [Fact]
+        public async Task GivenDifferentRequestor_WhenCreatingAnExportJob_ThenNewJobShouldBeCreated()
+        {
+            _claimsExtractor.ExtractImpl = () => new[] { KeyValuePair.Create("oid", "user1") };
+
+            var request = new CreateExportRequest(RequestUrl, DestinationType, ConnectionString);
+
+            CreateExportResponse response = await _createExportRequestHandler.Handle(request, _cancellationToken);
+
+            _claimsExtractor.ExtractImpl = () => new[] { KeyValuePair.Create("oid", "user1") };
```

Comment:
How are these different requestors? It looks like the claims extractor extracts the same thing. It looks like the connection string is what is changing.

Reply:
The pending change was accidentally not staged before I committed change.